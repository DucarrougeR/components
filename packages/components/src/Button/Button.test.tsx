import { createWithTheme, assertSnapshot } from 'looker-components-test-utils'
import 'jest-styled-components'
import React from 'react'
import { Button } from './Button'
import { ButtonSizes } from './size'

const noop = () => {}

test('Button default', () => {
  assertSnapshot(<Button>ğŸ¥‘</Button>)
})

test('Button variant outline', () => {
  assertSnapshot(<Button variant="outline">ğŸ¥‘</Button>)
})

test('Button variant transparent', () => {
  assertSnapshot(<Button variant="transparent">ğŸ¥‘</Button>)
})

test('Button type submit', () => {
  assertSnapshot(<Button type="submit">ğŸ¥‘</Button>)
})

test('Button type reset', () => {
  assertSnapshot(<Button type="reset">ğŸ¥‘</Button>)
})

test('Button type button', () => {
  assertSnapshot(<Button type="button">ğŸ¥‘</Button>)
})

test('Button type menu', () => {
  assertSnapshot(<Button>ğŸ¥‘</Button>)
})

test('Button padding', () => {
  assertSnapshot(<Button p="none">ğŸ¥‘</Button>)
})

test('Button padding', () => {
  assertSnapshot(
    <Button px="xxsmall" py="xxxlarge">
      ğŸ¥‘
    </Button>
  )
})

test('Button primary color', () => {
  assertSnapshot(<Button color="primary">ğŸ¥‘</Button>)
})

test('Button danger color', () => {
  assertSnapshot(<Button color="danger">ğŸ¥‘</Button>)
})

test('Button should accept disabled', () => {
  assertSnapshot(<Button disabled>ğŸ¥‘</Button>)
})

test('Button accepts a className prop', () => {
  const component = createWithTheme(<Button className="foobar">Hi</Button>)
  const tree = component.toJSON()
  if (!tree) throw new Error('component is NULL')
  expect(tree.props.className).toContain('foobar')
})

test('Button validates all sizes', () => {
  const sizes: ButtonSizes[] = ['xsmall', 'small', 'medium', 'large']
  sizes.forEach(size => {
    assertSnapshot(<Button size={size}>Test</Button>)
  })
})

test('Button allows autoFocus', () => {
  assertSnapshot(<Button autoFocus>Autofocus?</Button>)
})

test('Button allows for HTML events', () => {
  assertSnapshot(<Button onMouseEnter={noop}>Mouseenter?</Button>)
  assertSnapshot(<Button onClick={noop}>Click?</Button>)
})

test('Button allows for ARIA attributes', () => {
  assertSnapshot(<Button aria-disabled>aria-disabled</Button>)
})
