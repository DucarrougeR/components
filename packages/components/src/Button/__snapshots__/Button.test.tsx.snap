// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button allows autoFocus 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  autoFocus={true}
  className="c0 "
  size="medium"
>
  Autofocus?
</button>
`;

exports[`Button allows for ARIA attributes 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  aria-disabled={true}
  className="c0 "
  size="medium"
>
  aria-disabled
</button>
`;

exports[`Button allows for HTML events 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  onMouseEnter={[Function]}
  size="medium"
>
  Mouseenter?
</button>
`;

exports[`Button allows for HTML events 2`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  onClick={[Function]}
  size="medium"
>
  Click?
</button>
`;

exports[`Button danger color 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #CC1F36;
  border: 1px solid #CC1F36;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(204,31,54,0.25);
}

.c0:active,
.c0.active {
  background: #990F14;
  border-color: #990F14;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #B2121F;
  border-color: #B2121F;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #CC1F36;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  color="danger"
  size="medium"
>
  ðŸ¥‘
</button>
`;

exports[`Button default 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  size="medium"
>
  ðŸ¥‘
</button>
`;

exports[`Button padding 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
  padding: 0rem;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  size="medium"
>
  ðŸ¥‘
</button>
`;

exports[`Button padding 2`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
  padding-left: 0.25rem;
  padding-right: 0.25rem;
  padding-top: 3.75rem;
  padding-bottom: 3.75rem;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  size="medium"
>
  ðŸ¥‘
</button>
`;

exports[`Button primary color 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  color="primary"
  size="medium"
>
  ðŸ¥‘
</button>
`;

exports[`Button should accept disabled 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  disabled={true}
  size="medium"
>
  ðŸ¥‘
</button>
`;

exports[`Button type button 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  size="medium"
  type="button"
>
  ðŸ¥‘
</button>
`;

exports[`Button type menu 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  size="medium"
>
  ðŸ¥‘
</button>
`;

exports[`Button type reset 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  size="medium"
  type="reset"
>
  ðŸ¥‘
</button>
`;

exports[`Button type submit 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  size="medium"
  type="submit"
>
  ðŸ¥‘
</button>
`;

exports[`Button validates all sizes 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 0.75rem;
  height: 22px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  size="xsmall"
>
  Test
</button>
`;

exports[`Button validates all sizes 2`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 0.875rem;
  height: 26px;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  size="small"
>
  Test
</button>
`;

exports[`Button validates all sizes 3`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  size="medium"
>
  Test
</button>
`;

exports[`Button validates all sizes 4`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1.375rem;
  height: 42px;
  padding-left: 1.25rem;
  padding-right: 1.25rem;
  background: #64518A;
  border: 1px solid #64518A;
  color: #FFFFFF;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #473764;
  border-color: #473764;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

<button
  className="c0 "
  size="large"
>
  Test
</button>
`;

exports[`Button variant outline 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: #FFFFFF;
  border: 1px solid #C1C6CC;
  color: #64518A;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #64518A;
  border-color: #64518A;
  color: #FFFFFF;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #FFFFFF;
  border-color: #64518A;
  color: #2A1B60;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: #FFFFFF;
  border-color: #C1C6CC;
  color: #64518A;
}

<button
  className="c0 "
  size="medium"
>
  ðŸ¥‘
</button>
`;

exports[`Button variant transparent 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  vertical-align: middle;
  white-space: nowrap;
  font-size: 1rem;
  height: 34px;
  padding-left: 1rem;
  padding-right: 1rem;
  background: transparent;
  border: 1px solid transparent;
  color: #64518A;
}

.c0[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c0:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c0:active,
.c0.active {
  background: #E8E5FF;
  border-color: #E8E5FF;
  color: #6C43E0;
}

.c0:hover,
.c0:focus,
.c0.hover {
  background: #F3F2FF;
  border-color: #F3F2FF;
  color: #6C43E0;
}

.c0[disabled]:hover,
.c0[disabled]:active,
.c0[disabled]:focus {
  background-color: transparent;
  border-color: transparent;
  color: #64518A;
}

<button
  className="c0 "
  size="medium"
>
  ðŸ¥‘
</button>
`;
